### PR TITLE
unparam lint

### PR DESCRIPTION
**Pull request by mahmud2011**
#### Summary
* fix "always receives ..." lint err
* add unparam lint check

#### Ref
PR: https://github.com/mattermost/mattermost-server/pull/16653

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16622

#### Release Note
```release-note
None
```

### CODE REVIEW

File: app/file.go [hunk 1]
```diff
@@ -708,7 +708,7 @@ func (a *App) UploadFileX(channelId, name string, input io.Reader,
 
 	if *a.Config().FileSettings.DriverName == "" {
 		return nil, t.newAppError("api.file.upload_file.storage.app_error",
-			"", http.StatusNotImplemented)
+			"api.file.upload_file.storage.app_error", http.StatusNotImplemented)
```
Comment (agnivade) [inline app/file.go:711]: This seems to be unintentional. Let's revert this. That parameter is meant to be the error string. In the absence of an error string, an empty field is the correct way.

File: app/svg_test.go [hunk 2]
```diff
@@ -64,6 +64,7 @@ func TestParseInvalidSVGData(t *testing.T) {
 	invalidSVGs := []io.Reader{
 		generateSVGData(width, height, false, false, false), // missing viewBox, width & height
 		generateSVGData(width, 0, false, true, false),       // missing viewBox, malformed width & height
+		generateSVGData(0, height, false, true, false),      // missing viewBox, malformed width & height
```
Comment (agnivade) [inline app/svg_test.go:67]: This seems unrelated?

File: services/searchengine/bleveengine/bleve_test.go [hunk 3]
```diff
@@ -113,7 +114,7 @@ func (s *BleveEngineTestSuite) TestDeleteChannelPosts() {
 		channelToAvoidID := model.NewId()
 		posts := make([]*model.Post, 0)
 		for i := 0; i < 10; i++ {
-			post := createPost(userID, channelID, "test one two three")
+			post := createPost(userID, channelID, "test one two three "+time.Now().String())
```
Comment (agnivade) [inline services/searchengine/bleveengine/bleve_test.go:117]: unrelated?

**Review by agnivade — COMMENTED** [review]
Great work!

It looks like there are some additional changes which add new test coverage/fix flaky tests? Can we make them as part of a different PR? I would like this to be only about the `unparam` linter.

File: app/file.go (hunk 1 above)
Comment (mahmud2011) [reply app/file.go:711]: Actually you need to provide something in `details` otherwise you will get this error `always receives "" ...`. In every test case `details` is empty string.

**Comment by mahmud2011** [comment]
> Great work!
> 
> It looks like there are some additional changes which add new test coverage/fix flaky tests? Can we make them as part of a different PR? I would like this to be only about the `unparam` linter.

I only made changes to satisfy unparam lint nothing more.

File: app/svg_test.go (hunk 2 above)
Comment (mahmud2011) [reply app/svg_test.go:67]: In every case `width` is same that's why the error `always receives ...`.

File: services/searchengine/bleveengine/bleve_test.go (hunk 3 above)
Comment (mahmud2011) [reply services/searchengine/bleveengine/bleve_test.go:117]: In every case `"test one two three"`

File: app/file.go (hunk 1 above)
Comment (agnivade) [reply app/file.go:711]: Oh I see. Let's remove the `details` parameter then.

File: services/searchengine/bleveengine/bleve_test.go (hunk 3 above)
Comment (agnivade) [reply services/searchengine/bleveengine/bleve_test.go:117]: Let's use `model.NewRandomString` for this.

File: app/svg_test.go (hunk 2 above)
Comment (agnivade) [reply app/svg_test.go:67]: Let's add an ignore statement for this now, and fix this in a separate PR.

**Comment by mahmud2011** [comment]
services/mailservice/mail.go:260:69: `sendMailUsingConfigAdvanced` - `enableComplianceFeatures` is unused (unparam)
func sendMailUsingConfigAdvanced(mail mailData, config *SMTPConfig, enableComplianceFeatures bool) error {

Local golangci-lint does not catch this. -_-

**Review by agnivade — APPROVED** [review]
Great work!

**Comment by agnivade** [comment]
This is pending changes in the EE codebase too. I will take care of it.

File: services/searchengine/bleveengine/bleve_test.go [hunk 4]
```diff
@@ -138,7 +138,7 @@ func (s *BleveEngineTestSuite) TestDeleteChannelPosts() {
 		userID := model.NewId()
 		channelID := model.NewId()
 		channelToDeleteID := model.NewId()
-		post := createPost(userID, channelID, "test one two three")
+		post := createPost(userID, channelID, model.NewRandomString(15))
```
Comment (streamer45) [inline services/searchengine/bleveengine/bleve_test.go:141]: I think here (and in the other related cases) the best approach would be to remove the parameter and generate the random message directly inside `createPost`.

**Review by streamer45 — APPROVED** [review]
Thank you @mahmud2011 :+1: 
Left one minor suggestion but otherwise looks good!

File: services/searchengine/bleveengine/bleve_test.go (hunk 4 above)
Comment (agnivade) [reply services/searchengine/bleveengine/bleve_test.go:141]: @mahmud2011 - can you please look into this?

**Comment by agnivade** [comment]
Hey @mahmud2011 - Just a small thing needed to be done here: https://github.com/mattermost/mattermost-server/pull/16927#discussion_r589964103.

File: services/searchengine/bleveengine/bleve_test.go (hunk 4 above)
Comment (mahmud2011) [reply services/searchengine/bleveengine/bleve_test.go:141]: @agnivade Sorry, I missed it. Happy to do that.
